clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -401,7 +404,6 @@ public void commitAsync(final Map<TopicIdPartition, Acknowledgements> acknowledg
 
                         metricsManager.recordAcknowledgementSent(acknowledgements.size());
                         log.debug("Added async acknowledge request for partition {} to node {}", tip.topicPartition(), node.id());
-                        resultCount.incrementAndGet();
```

Comment:
This is a bit weird, making and then not using the AtomicInteger. I expect with a small code change, you could safely not use the result count for async commits.

Reply:
Right yeah, I have removed the result count for commitAsync now.